lib/mock-axios-types.ts

```diff
@@ -149,6 +149,20 @@ export interface AxiosMockAPI {
      */
     getReqByMatchUrl: (url: RegExp) => AxiosMockQueueItem;
 
+    /**
+     * Returns the most recent request matching any key with the regex (e.g.: url, data, config)
+     *   Can also use multiple keys for research. For example: getReqByMatch({ url: /batch/, data: /employees/ })
+     *
+     * Returns undefined if no matching request could be found
+     *
+     * The result can then be used with @see mockResponse
+     *
+     * @param opts { key_a: RegExp_a, ..., key_n: RegExp_n }
+     *          key_x: The key of the request to be found
+     *          RegExp_x: The RegExp to be tested against that key
+     */
+    getReqByMatch: (opts: object) => AxiosMockQueueItem;
```

Comment:
Can we use a more specific parameter type here than just `object`? Maybe something like ` {[key in keyof AxiosMockQueueItem]+?: RegExp}` would work? (Same for the actual implementation)

Reply:
Hmm, this is the first _TypeScript_ code I'm coding so you should be right about these parameter types :)

I'll use that

```rb
{[key in keyof AxiosMockQueueItem]+?: RegExp}
```

👍

Reply:
Done!